### PR TITLE
ENH: adds `cluster_features_closed_reference`

### PR DESCRIPTION
**Pull request by gregcaporaso**
Fixes #17.

### CODE REVIEW

File: q2_vsearch/_cluster_features.py [hunk 1]
```diff
@@ -44,6 +44,9 @@ def _collapse_f_from_uc(uc):
             else:
                 pass
 
+    if len(id_to_centroid) == 0:
```
Comment (thermokarst) [inline q2_vsearch/_cluster_features.py:47]: 👍

File: q2_vsearch/_cluster_features.py [hunk 2]
```diff
@@ -72,26 +75,26 @@ def _fasta_with_sizes(input_fasta_fp, output_fasta_fp, table):
...
 
-    table_ids = set(table_ids)
-    non_overlapping_ids = table_ids - sequence_ids
-    if len(non_overlapping_ids) != 0:
-        raise ValueError('Some feature ids are present in table, but not in '
-                         'sequences. The set of features in sequences must be '
-                         'identical to the set of features in table. Feature '
-                         'ids present in table but not sequences are: %s'
-                         % ', '.join(non_overlapping_ids))
+    _error_on_nonoverlapping_ids(set(table_ids), sequence_ids,
```
Comment (thermokarst) [inline q2_vsearch/_cluster_features.py:78]: Cool, this change makes sense to me! Love the util func.

File: q2_vsearch/plugin_setup.py [hunk 3]
```diff
@@ -66,3 +71,60 @@
...
+                    'will launch one thread per CPU core.')
+    },
+    output_descriptions={
+        'clustered_table': 'The table following clustering of features.',
+        'unmatched_sequences': ('The sequences which failed to match any '
+                                'reference sequences. This output maps to '
+                                'vsearch\'s --notmatched parameter.')
+    },
+    name='Closed-reference clustering of features.',
+    description=('Given a feature table and the associated feature '
```
Comment (thermokarst) [inline q2_vsearch/plugin_setup.py:116]: 👍

File: q2_vsearch/tests/test_cluster_features.py [hunk 4]
```diff
@@ -88,7 +89,7 @@ def test_99_percent_clustering(self):
         exp_seqs = [exp_seqs[0], exp_seqs[3], exp_seqs[1]]
         self.assertEqual(obs_seqs, exp_seqs)
 
-    def test_97_percent_clustering(self):
+    def test_97_percent_clustering_feature1_most_abundant(self):
```
Comment (thermokarst) [inline q2_vsearch/tests/test_cluster_features.py:92]: Good catch.

File: q2_vsearch/tests/test_cluster_features.py [hunk 5]
```diff
@@ -266,6 +267,251 @@ def test_no_overlapping_feature_ids(self):
...
+        self.assertEqual(obs_table, exp_table)
+
+        obs_seqs = list(skbio.io.read(str(unmatched_seqs),
+                        constructor=skbio.DNA, format='fasta'))
+        exp_seqs = list(skbio.io.read(str(input_sequences),
+                        constructor=skbio.DNA, format='fasta'))
+        exp_seqs = [exp_seqs[1], exp_seqs[2]]
+        self.assertEqual(obs_seqs, exp_seqs)
+
+    def test_100_percent_clustering_strand(self):
```
Comment (thermokarst) [inline q2_vsearch/tests/test_cluster_features.py:308]: 👍

**Review by thermokarst — APPROVED** [review]
This looks great to me, thanks @gregcaporaso for taking the time to do a detailed review with me!

**Comment by thermokarst** [comment]
@gregcaporaso, if my [last commit](https://github.com/qiime2/q2-vsearch/pull/18/commits/e23361ae347c596ee4905fe7659fbc8cc4b4bb5f) looks good to you just let me know, this PR is ready for merge!

**Comment by gregcaporaso** [comment]
That last commit looks good @thermokarst, I also resolved the merge conflict (came up after merge of #19). This should be ready for merge!

**Pull request merged**